### PR TITLE
Update common files.

### PR DESCRIPTION
**Pull request by geeknoid**

### CODE REVIEW

File: common/scripts/lint_go.sh [hunk 1]
```diff
@@ -19,9 +21,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-updatecommon:
-	@git clone https://github.com/istio/common-files
-	@cd common-files
-	@git rev-parse HEAD >.commonfiles.sha
-	@cp -r common-files/files/* common-files/files/.[^.]* .
-	@rm -fr common-files
+golangci-lint run -j 8 -c ./common/config/.golangci.yml
```
Comment (howardjohn) [inline common/scripts/lint_go.sh]: this may fail with OOM. We had GOGC=25 set before which makes the GC more aggressive

**Comment by geeknoid** [comment]
@howardjohn Yes, I ran "go generate ./...". This resulted in slightly different proto descriptors due to different import paths. This in turn results in different yaml, since the yamls encoded the deacriptors as base 64.

**Comment by geeknoid** [comment]
The non-generic solution for .PHONY is to add the relevant definitions to Makefile.overrides.mk. That makefile is evaluated along with the top-level Makefile before launching the container. I think this can be done as a follow-on.

Which are the significant targets which wouldn't work with this change? It's limited by the set of top-level files/dirs at the root of istio/istio, right?

File: common/scripts/lint_go.sh (hunk 1 above)
Comment (geeknoid) [reply common/scripts/lint_go.sh]: Possible. I'll figure that out specifically in the next step when I start to leverage the new build logic, including the standardized linter suites.

Comment (howardjohn) [reply common/scripts/lint_go.sh]: oh right this isn't used yet. in my other pr I tried to get rid of it and got oom immediately which is why I was surprised this didnt

**Comment by howardjohn** [comment]
> Which are the significant targets which wouldn't work with this change? It's limited by the set of top-level files/dirs at the root of istio/istio, right?

I think so.. maybe we can just dynamically set phony based on all too level files. Or just do it statically with the current set of files

**Comment by geeknoid** [comment]
@howardjohn OK, I added a bunch of phony targets to Makefile.overrides.mk.

**Comment by geeknoid** [comment]
PTAL

**Comment by howardjohn** [comment]
```
$ make istioctl
make: Nothing to be done for 'istioctl'.
```

Seems like the phony in override doesn't work?

**Comment by geeknoid** [comment]
Even if I add a PHONY: istioctl in the top-level Makefile, I still get that
message...

**Comment by geeknoid** [comment]
OK, I found something that works, by adding explicit targets in
Makefile.override.mk. Please check it out again.

**Comment by howardjohn** [comment]
Wow I don't get what makefile is doing...

I tried to replace what you did with 
```
PHONYS := $(shell ls)
.PHONY: $(PHONYS)
$(PHONYS):
	@${MAKE} $@
```

and i get the 
```
make[1]: Nothing to be done for 'Makefile.overrides.mk'.
make[1]: Nothing to be done for 'Makefile'.
```

stuff again

**Comment by howardjohn** [comment]
Interesting, I think the problem is that we get a target for `Makefile.override.mk`. So if we replace with 
```
PHONYS := $(shell ls | grep -v Makefile)
.PHONY: $(PHONYS)
$(PHONYS):
	@$(MAKE) $@
```

It works?

Not sure if we need even more crazy makefile stuff though

**Comment by sdake** [comment]
> Interesting, I think the problem is that we get a target for `Makefile.override.mk`. So if we replace with
> 
> ```
> PHONYS := $(shell ls | grep -v Makefile)
> .PHONY: $(PHONYS)
> $(PHONYS):
> 	@$(MAKE) $@
> ```
> 
> It works?
> 
> Not sure if we need even more crazy makefile stuff though

this is actually a pretty cool solution - with appropriate comments.

**Comment by geeknoid** [comment]
@howardjohn Is this hack for Makefile or Makefile.overrides.mk?

**Comment by howardjohn** [comment]
Makefile.overrides.mk is how I tested it. It could work for Makefile
possibly? I did not try that though

**Comment by geeknoid** [comment]
OK, I've merge John's code in Makefile.overrides.mk. If we find we need this kind of cleverness in other repos, we can move it to Makefile in the common repo instead. But for now, let's keep it local to Makefile.overrides.mk in case it starts to show some problems...

File: bin/codecov.sh [hunk 2]
```diff
@@ -41,14 +41,14 @@ COVERAGEDIR="$(mktemp -d /tmp/istio_coverage.XXXXXXXXXX)"
 mkdir -p "$COVERAGEDIR"
 
 function cleanup() {
-  make localTestEnvCleanup
+  make -f Makefile.core.mk localTestEnvCleanup
```
Comment (howardjohn) [inline bin/codecov.sh]: Why do we have to the `-f`, is it so we don't do docker-in-docker if we are using container?

**Comment by geeknoid** [comment]
Correct.

File: bin/codecov.sh (hunk 2 above)
Comment (geeknoid) [reply bin/codecov.sh]: Correct.

**Pull request merged**